### PR TITLE
[modal] Tweaks and enhanced documentation

### PR DESCRIPTION
**Pull request by tmorehouse**
Add fix/feature for issue #521, and expands documentation a bit.

### CODE REVIEW

File: docs/components/modal/README.md [hunk 1]
```diff
@@ -2,11 +2,28 @@
...
-### Toggle Modal
+Modals, by default, have an **OK** and a **Close** button in the footer. These buttons can
+be cusomized by setting various props on the component. You can cusomize the size of the buttons,
+disable the **OK** button, hide the **Close** button (i.e. OK Only), and provide custom
+button content using the `ok-title` and `close-title` props, or using the named
+slots `modal-ok` and `modal-close`.
 
-##### Using `v-b-modal` directive (recommended)
+`b-modal` supports close on ESC (enabled by default), close on backdrop click (enabled by default), and 
+the `X` close button in the header (by default). HTese features may be disabled by setting the the
```
Comment (alexsasharegan) [inline docs/components/modal/README.md]: Little typo on line 14

**Comment by tmorehouse** [comment]
I think I'm getting documentation happy when I work on components, lol.  While it is fresh in my head I tend to want to write it out (complete with Troy-o's, lol).

File: docs/components/modal/README.md (hunk 1 above)
Comment (tmorehouse) [inline docs/components/modal/README.md]: De-troy-o'd

**Comment by tmorehouse** [comment]
The extra documentation should help with issues like #462

**Pull request merged**